Assets/MixedRealityToolkit/Utilities/MixedRealityToolkitSceneChecker.cs

```diff
@@ -0,0 +1,130 @@
+                var managerSearch = Object.FindObjectsOfType<MixedRealityToolkit>();
+
+                if (managerSearch.Length == 0)
+                {
+                    HideNoActiveToolkitWarning = SessionState.GetBool(HideNoActiveToolkitWarningKey, false);
+                    if (!HideNoActiveToolkitWarning)
+                    {
+                        NoActiveToolkitWarning.OpenWindow(null);
+                    }
+                    return; 
+                }
+            }
+        }
+
+        internal class NoActiveToolkitWarning : EditorWindow
+        {
+            private static NoActiveToolkitWarning activeWindow;
+            [SerializeField]
+            public MixedRealityToolkitConfigurationProfile configurationProfile;
+            private bool hideWarning = false;
+
+            public static void OpenWindow(MixedRealityToolkitConfigurationProfile configurationProfile)
+            {
+                // If we already have an active window, bail
+                if (activeWindow != null)
+                {
+                    return;
+                }
+
+                activeWindow = EditorWindow.GetWindow<NoActiveToolkitWarning>();
+                activeWindow.configurationProfile = configurationProfile;
+                activeWindow.maxSize = new Vector2(400, 140);
+                activeWindow.minSize = new Vector2(400, 140);
+                activeWindow.titleContent = new GUIContent("No Active Toolkit Found");
+
+                activeWindow.ShowPopup(); 
+            }
+
+            private void OnGUI()
+            {
+                EditorGUILayout.HelpBox("There is no active Mixed Reality Toolkit in your scene. Would you like to create one now?", MessageType.Warning);
+
+                hideWarning = EditorGUILayout.Toggle("Don't show this again", hideWarning);
+
+                configurationProfile = (MixedRealityToolkitConfigurationProfile)EditorGUILayout.ObjectField(configurationProfile, typeof(MixedRealityToolkitConfigurationProfile), false);
+                if (configurationProfile == null)
+                {
+                    EditorGUILayout.HelpBox("Select a configuration profile.", MessageType.Info);
+                }
+
+                EditorGUILayout.BeginHorizontal();
+                GUI.enabled = (configurationProfile != null);
+                if (GUILayout.Button("Yes"))
+                {
+                    var playspace = MixedRealityToolkit.Instance.MixedRealityPlayspace;
+                    Debug.Assert(playspace != null);
+                    MixedRealityToolkit.Instance.ActiveProfile = configurationProfile;
+
+                    SessionState.SetBool(HideNoActiveToolkitWarningKey, hideWarning);
+                    Close();
+                }
+                GUI.enabled = true;
+
+                if (GUILayout.Button("No"))
+                {
+                    SessionState.SetBool(HideNoActiveToolkitWarningKey, hideWarning);
```

Comment:
I think this setting should be a settings scriptable object somewhere.

That way the variable is persisted in a config object, rather than in an ephemeral session which is cleared on each editor boot.

Moroever, using EditorPrefs could make it difficult to re-enable if need be.

Reply:
We could use [EditorPrefs](https://docs.unity3d.com/ScriptReference/EditorPrefs.html).